packages/hix/lib/Hix/Monad.hs

```diff
@@ -271,3 +281,13 @@ appContextDebug desc ma = do
     logMsg = case Text.uncons desc of
       Just (h, t) -> Text.cons (toUpper h) t
       Nothing -> desc
+
+resolvePathSpecDir :: PathSpec Dir -> M (Path Abs Dir)
+resolvePathSpecDir pathSpec = do
+  AppResources {cwd} <- ask
+  liftE $ resolvePathSpec resolveDir cwd pathSpec
+
+resolvePathSpecFile :: PathSpec File -> M (Path Abs File)
+resolvePathSpecFile pathSpec = do
+  AppResources {cwd} <- ask
+  liftE $ resolvePathSpec resolveFile cwd pathSpec
```

Comment:
```suggestion
  appContext [exon|resolving path spec #{showP pathSpec}|] do
    AppResources {cwd} <- ask
    liftE $ resolvePathSpec resolveFile cwd pathSpec
```

In case the resolver throws some obscure IO error.

This'll require `instance Pretty (PathSpec a)`

Reply:
yeah that's fine, that all grew out of laziness and necessity anyway